baremetal/metal3datatemplate_manager.go

```diff
@@ -135,17 +136,12 @@ func (m *DataTemplateManager) getIndexes(ctx context.Context) (map[int]string, e
 			continue
 		}
 
-		// Get the claim Name, if unset use empty string, to still record the
-		// index being used, to avoid conflicts
-		claimName := ""
-		if dataObject.Spec.Claim.Name != "" {
-			claimName = dataObject.Spec.Claim.Name
-		}
-		m.DataTemplate.Status.Indexes[claimName] = dataObject.Spec.Index
+		claimName := dataObject.Spec.Claim.Name
+		statusIndexes[claimName] = dataObject.Spec.Index
 		indexes[dataObject.Spec.Index] = claimName
```

Comment:
`indexes` and `statusIndexes` maps are the same, in case `indexes` `dataObject.Spec.Index` is key and value is`claimName` and `statusIndexes` has `claimName` as key and `dataObject.Spec.Index` as value. Can we not achieve what we are trying to do just by `indexes`?

Reply:
They are "co-maps" if you want to access from index or from claims you will not use the same map. 
* statusIndexes "replaces" m.DataTemplate.Status.Indexes. We use it to get a filtered Indexes where we only keep those for which a claim exists
* indexes is in fact used as a set (value never really used). It is used to find which index can be used when a Data is created.
We could defer the creation of the indexes map. If there is no Data creation it would be faster but unless we are careful it may be slower if there are multiple Data created. This is an optimization (or not) unrelated to the issue.